### DISCUSSION TITLE
Lesson 11: Gas estimation failed for Rinkeby test

### DISCUSSION THREAD

**Discussion by nhache** [smartcontractkit/full-blockchain-solidity-course-py#177] [Q&A] [answered]
At 10h:50ish mins we test that `deploy_and_create()` works with Rinkeby before writing tests.

- Contract creation works (tx: 0xb1dcf038212a275693362ae8c3ef1265e273a88d07c77353b01bb0fd0e534918)
- Link funding works (tx: 0xb1dcf038212a275693362ae8c3ef1265e273a88d07c77353b01bb0fd0e534918)

Then I receive 
`ValueError: Gas estimation failed: 'execution reverted: Below agreed payment'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.`

My test ETH account has enough ETH even though I am confused that there is a mention of an agreed payment (our contract AdvancedCollectible isn't payable)

**Comment by nhache** [comment, marked as answer]
Always pleasant when you manage to answer your own question. I had a hard to debug error in `AdvancedCollectible.sol`, assigning `fee` in the `constructor` to itself instead of the parameter `_fee` ...

**Reply by PatrickAlphaC** [reply]
Boom! Way to stay determined!
